notebooks/Notebook3.md

```diff
@@ -2,33 +2,22 @@
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+<p>I started working on [my parser](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/DSL%20Code/parser.py) this week, so I also made adjustments to my language design to make it so that my parser could get to my desired result better. So far making the parser doesn't seem too hard. I was able to output an extendscript file that selected a composition to work on. Currently I'm able to process code that selected the current composition that is selected in after effects or that pics a specific composition with a given index. </p>
+
+<p>This is the [output I've gotten so far](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/select_comp.png) from my parser, and this is my [goal output](https://github.com/hmc-cs111-spring2023/Artifact-Cruedy/blob/main/goal_output.png). I'm going to make sure that my parser can catch errors, I already added an error catch that tells the user to make sure that their code picks a composition in the first line of code.</p>
+
+<p>I am nervous that the way that I've been writing my parser is incorrect because it seems to simple or that the way I've written my parser is too prone to mistakes. If I find out that I've done it the wrong way, it's going to push back my progress. I'm treating the file that that I write my DSL in as a txt file, and that text file is just read into the parser line by line. When it reads those lines in, it just translates it to extenscript.</p>
 
 ## Questions
 
-**What is the most pressing issue for your project? What design decision do
-you need to make, what implementation issue are you trying to solve, or how
-are you evaluating your design and implementation?**
+I feel like I'm making the parser wrong. Since I just have to read in one file and put out a javascript file from it, it seems like I'm pretty much just hard coding what each line.
 
-**What questions do you have for your critique partners? How can they best help
-you?**
+Am I parsing it wrong if this is the way I'm doing it? It kinda feels similar to the example piconot parser because that one just looked at characters and looked at what to do for each case.
```

Comment:
A parser doesn't always need to be as complicated as the most complicated parser that we have encountered in the class! If the syntax is relatively simple (e.g., no nesting expressions), the parser can also be relatively simple. 

We can chat more about the parser, if you'd like! A quick 10 minutes (or evern less) might help you feel good about your design and / or identify some changes to the design.